packages/protocol/contracts/AirnodeRrp.sol

```diff
@@ -1,239 +1,161 @@
-    /// @param fulfillFunctionId Signature of the function that will be called
-    /// to fulfill
-    /// @return callSuccess If the fulfillment call succeeded
-    /// @return callData Data returned by the fulfillment call (if there is
-    /// any)
-    function fulfill(
-        bytes32 requestId,
-        bytes32 airnodeId,
-        uint256 statusCode,
-        bytes calldata data,
-        address fulfillAddress,
-        bytes4 fulfillFunctionId
-    )
+    /// @notice Called to get the Airnode announcement
+    /// @dev The information announced with this function is not trustless.
+    /// It is up to the user to verify that the announced `xpub` is correct by
+    /// checking if its default BIP 44 wallet matches the Airnode address.
+    /// It is not possible to verify the correctness of `authorizers` (i.e.,
+    /// that the Airnode will use these contracts to check for authorization).
+    /// @param airnode Airnode address
+    /// @return xpub Extended public key of the Airnode
+    /// @return authorizers Authorizer contract addresses
+    function getAirnodeAnnouncement(address airnode)
         external
+        view
         override
-        onlyCorrectFulfillmentParameters(
-            requestId,
-            airnodeId,
-            fulfillAddress,
-            fulfillFunctionId
-        )
-        returns (bool callSuccess, bytes memory callData)
+        returns (string memory xpub, address[] memory authorizers)
     {
-        delete requestIdToFulfillmentParameters[requestId];
-        emit ClientRequestFulfilled(airnodeId, requestId, statusCode, data);
-        (callSuccess, callData) = fulfillAddress.call( // solhint-disable-line
-            abi.encodeWithSelector(
-                fulfillFunctionId,
-                requestId,
-                statusCode,
-                data
-            )
-        );
+        Announcement storage announcement = airnodeToAnnouncement[airnode];
+        return (announcement.xpub, announcement.authorizers);
     }
 
-    /// @notice Called by Airnode if the request cannot be fulfilled
-    /// @dev Airnode should fall back to this if a request cannot be fulfilled
-    /// because fulfill() reverts
-    /// @param requestId Request ID
-    /// @param airnodeId Airnode ID from AirnodeParameterStore
-    /// @param fulfillAddress Address that will be called to fulfill
-    /// @param fulfillFunctionId Signature of the function that will be called
-    /// to fulfill
-    function fail(
-        bytes32 requestId,
-        bytes32 airnodeId,
-        address fulfillAddress,
-        bytes4 fulfillFunctionId
-    )
+    /// @notice A convenience method to retrieve multiple templates with a
+    /// single call
+    /// @param templateIds Request template IDs
+    /// @return airnodes Array of Airnode addresses
+    /// @return endpointIds Array of endpoint IDs
+    /// @return parameters Array of request parameters
+    function getTemplates(bytes32[] calldata templateIds)
```

Comment:
I had it there because it's optional / a convenience function because the node may just as well use `templates()` to get the templates individually. I will reorganize everything in a way that all templates stuff is together though.